sympy/holonomic/holonomic.py

```diff
@@ -520,7 +520,7 @@ def _singularics_to_ord(self):
             y0 = []
             for i in range(a):
                 y0.append(S(0))
-            y0 += b
+            y0 += [j * factorial(a + i) for i, j in enumerate(b)]
```

Comment:
The condition may be satisfied even if  a  is a floating point number. It should be cast to an integer in the argument. (`==` means mathematical equality, the objects may different.)